### PR TITLE
Add support for quantized multiply to Relay

### PR DESCRIPTION
**Pull request by ekalda**
This patch adds multiply operator for quantized tensors. The details of the quantized multiplication are outlined in the code.

This builds on pull request 3927 and includes the changes Animesh mentions in the comments on that request.

### CODE REVIEW

**Comment by ekalda** [comment]
Hi @anijain2305, that's the new version of QNN multiply (Tristan is on holiday for the next few weeks), can you review it? Thank you!

File: python/tvm/relay/qnn/op/qnn.py [hunk 1]
```diff
@@ -349,3 +349,44 @@ def dense(data,
...
+    Parameters
+    ----------
+    lhs : relay.Expr
+        The left hand side quantized input data.
+
+    rhs : relay.Expr
+        The right hand side quantized input data.
+
+    lhs_scale: float
+        The scale of the lhs quantized expr.
```
Comment (anijain2305) [inline python/tvm/relay/qnn/op/qnn.py]: One more blank line here.

File: tests/python/relay/test_qnn_mul.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    z = relay.qnn.op.mul(
+        lhs=x,
+        rhs=y,
+        lhs_scale=lhs_scale,
+        lhs_zero_point=lhs_zero_point,
+        rhs_scale=rhs_scale,
+        rhs_zero_point=rhs_zero_point,
+        output_scale=output_scale,
+        output_zero_point=output_zero_point,
+    )
```
Comment (anijain2305) [inline tests/python/relay/test_qnn_mul.py]: Align.

File: tests/python/relay/test_qnn_mul.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
```
Comment (anijain2305) [inline tests/python/relay/test_qnn_mul.py]: Align

File: src/relay/qnn/op/mul.cc [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+A tensor multiplication c = a * b can be written in terms of respective
+quantized tensors, scales and zero points as
+S_c * (Q_c - zp_c) = S_a * (Q_a - zp_a) * S_b * (Q_b - zp_b).
+
+We can consider the product (Q_a - zp_a) * (Q_b - zp_b) as a different
+quantized tensor of c, Q', with corresponding scale S' = S_a * S_b and zp' =
+0. The quantized multiplication then becomes
+Q_c = S'/S_c Q' + z_c,
+which is essentially a requantization of tensor Q' into tensor Q_c.
+*/
```
Comment (anijain2305) [inline src/relay/qnn/op/mul.cc]: Minor - 1) Remove first blank line, 2) Align the comments.

**Comment by ekalda** [comment]
Thank you for your comments, I fixed the indentation and force-pushed it again.

**Comment by anijain2305** [comment]
@jackwish Can you please review it once as well?

**Comment by ekalda** [comment]
Re-triggered the CI again, everything passed successfully.

**Pull request merged**